app/models/backup_code_configuration.rb

```diff
@@ -0,0 +1,54 @@
+class BackupCodeConfiguration < ApplicationRecord
+  include EncryptableAttribute
+
+  devise
+
+  include EncryptableAttribute
+
+  encrypted_attribute_without_setter(name: :code)
+
+  # IMPORTANT this comes *after* devise() call.
+  include BackupCodeEncryptedAttributeOverrides
+
+  belongs_to :user
+
+  def self.unused
+    where(used: false)
+  end
+
+  def mfa_enabled?
+    used == false
+  end
+
+  # This method smells of :reek:UtilityFunction
+  def selection_presenters
+    [TwoFactorAuthentication::BackupCodeSelectionPresenter.new]
+  end
+
+  def friendly_name
+    :backup_codes
+  end
+
+  def self.selection_presenters(set)
+    if set.any?
+      set.first.selection_presenters
+    else
+      []
+    end
+  end
+
+  class << self
+    def find_with_code(code:, user_id:)
+      return nil if !code.is_a?(String) || code.empty?
```

Comment:
Style: the `nil` is superfluous here since that's what gets returned by a bare `return`. I'd be more comfortable with
`return unless code.present?`

If we're not passing a string into the `find_with_code` method, that's a bug we need to track down elsewhere.